### DISCUSSION TITLE
[Dynamic] apply the impulse (BCs) on the nodes or the elements level instead of surface?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#20121] [Q&A Modules: Solid mechanics] [answered]
Hi guys,

Just wondering if it is possible to apply the impulse on nodes or elements level? I see from the example, like this one `dynamics/prescribed_displacement/3D_QStatic_1_Ramped_Displacement_with_gravity.i`, we can apply BCs [PresetAcceleration PresetDisplacement PresetVelocity] only on the surface level.

```
[Functions]
  [./displacement_front]
    type = PiecewiseLinear
    data_file = 'displacement.csv'
    format = columns
  [../]
[]

[BCs]
  [./prescribed_displacement]
    type = PresetDisplacement
    variable = disp_z
    velocity = vel_z
    acceleration = accel_z
    beta = 0.25
    boundary = front
    function = displacement_front
  [../]

```
Is there any way to work around this?

Kind regards,
Traiwit

**Comment by GiudGiud** [comment, marked as answer]
@jiangwen84

**Reply by GiudGiud** [reply]
@Traiwit still need help on this?
I can tag others who may know

**Reply by Traiwit** [reply]
Hi @GiudGiud I kinda know how to work around it using external software to create surfaces around particular nodes/elements, but yeh it would be nice to hear from the expert if it is possible to do this without having to generate extra surfaces.

Thanks!
Traiwit

**Reply by GiudGiud** [reply]
@dschwen @lindsayad

**Reply by lindsayad** [reply]
In theory we could develop classes that apply displacements for specific node ids but I don't know if this is widely desired thing. Picking out node ids is often a cumbersome process, especially for large number of nodes. Consequently choosing to apply for sidesets/nodesets is typically more desirable for our users. We do have the `ExtraNodesetGenerator` that allows you to create a nodeset from specific nodes.

Let me know if I have misunderstood what you want.

**Reply by Traiwit** [reply]
Hi @lindsayad,

Applying BCs on nodesets is at least better than having to generate surfaces/sidesets.

BUT is it possible to do that? for example `PresetDisplacement` requires boundary/side, or do you mean use `ExtraNodesetGenerator` then convert that into boundary/side?

![image](https://user-images.githubusercontent.com/79562439/157140281-591ab11b-c74a-4c95-8d1e-90d6d2136ef9.png)

Kind regards,
Traiwit

**Reply by lindsayad** [reply]
Use `ExtraNodesetGenerator` to create a new nodeset and then whatever you specified for `new_boundary` in the `ExtraNodesetGenerator` should be your argument to `boundary` in `PresetDisplacement`

**Reply by Traiwit** [reply]
works perfectly, thank you @lindsayad and @GiudGiud
